### ISSUE TITLE
Inconsistent ability to read/write cells dependent on whether file was open already

### ISSUE THREAD

**Issue by UlamNajk** [ZoomerAnalytics/xlwings#761]
Using sheet["A1"].value and sheet[0, 0].value inconsistently give me errors.
When the file in question is open before the xlwings program is run, the program runs and completes with no errors but if the workbook is not open and is opened via the xlwings Book($PATH$) command then the error occurs. Traceback is as follows:
The code in question is
```
    # Open file, summary sheet as data objects
    the_app = xw.App()
    the_app.api.AskToUpdateLinks = False
    the_app.api.DisplayAlerts = False

    # Try to manually open all Bloomberg extensions
    xla1 = xw.Book(r'C:\blp\API\Office Tools\BloombergUI.xla')
    xla2 = xw.Book(r'C:\blp\API\Office Tools\BloombergUIChartsEvents.xla')
    the_app.api.RegisterXLL(r'C:\blp\API\Office Tools\blpxladdin.xll')
    the_app.api.RegisterXLL(r'C:\blp\API\Office Tools\brtdwrap.xll')
    xla1.api.RunAutoMacros(1)
    xla2.api.RunAutoMacros(1)

    wrkbk_file = xw.Book(active_file)
    summ = wrkbk_file.sheets("Summary")

    time.sleep(10)

    summ['B1'].value = date.today() + timedelta(days=-1)
```
and the traceback is 
```
Traceback (most recent call last):
  File "s:/.../xlwings_test.py", line 626, in <module>
    main(False)
  File "s:/.../xlwings_test.py", line 71, in main
    summ['B1'].value = date.today() + timedelta(days=-1)
  File "C:\Users\XXX\AppData\Local\Continuum\Anaconda2\lib\site-packages\xlwings\main.py", line 919, in __getitem__
    return self.range(item)
  File "C:\Users\XXX\AppData\Local\Continuum\Anaconda2\lib\site-packages\xlwings\main.py", line 820, in range
    return Range(impl=self.impl.range(cell1, cell2))
  File "C:\UsersXXX\AppData\Local\Continuum\Anaconda2\lib\site-packages\xlwings\_xlwindows.py", line 583, in range
    xl1 = self.xl.Range(arg1)
  File "C:\Users\XXX\AppData\Local\Continuum\Anaconda2\lib\site-packages\win32com\client\dynamic.py", line 527, in __getattr__
    raise AttributeError("%s.%s" % (self._username_, attr))
AttributeError: <unknown>.Range
```

I can't fathom why the ability to read or to write from a cell would be inconsistent.  Any help is appreciated.

**Comment by UlamNajk** [comment]
Removing the lines concerning .xla and .xll files seems to have fixed the read/write problem, but prevents the workbook from updating the data links properly.

**Comment by fzumstein** [comment]
can you check if you get the same issues when you use
`the_app.books.open(...)` instead of `xw.Book`?

Am 27.09.2017 8:43 nachm. schrieb "Liam Kane" <notifications@github.com>:

**Comment by UlamNajk** [comment]
I should convert the_app to be xw.apps[0] for this as well, right?

**Comment by fzumstein** [comment]
probably not: `xw.App` launches a new Excel instance and `xw.apps[0]` takes
the last active instance but expects a running instance

Am 27.09.2017 9:21 nachm. schrieb "Liam Kane" <notifications@github.com>:

**Comment by UlamNajk** [comment]
Hesitant to test because Bloomberg addin disappears from ribbon, apparently only after trying to run this code.  Trying to resolve that first, might open a new issue.  It's similar to [this issue](https://github.com/ZoomerAnalytics/xlwings/issues/608) which is what led me to have the 
```
# Try to manually open all Bloomberg extensions
    xla1 = xw.Book(r'C:\blp\API\Office Tools\BloombergUI.xla')
    xla2 = xw.Book(r'C:\blp\API\Office Tools\BloombergUIChartsEvents.xla')
    the_app.api.RegisterXLL(r'C:\blp\API\Office Tools\blpxladdin.xll')
    the_app.api.RegisterXLL(r'C:\blp\API\Office Tools\brtdwrap.xll')
    xla1.api.RunAutoMacros(1)
    xla2.api.RunAutoMacros(1)
```
code block in the first place.

**Comment by UlamNajk** [comment]
Using the_app.books.open() did fix the read/write problem.
